pinot-broker/src/test/java/com/linkedin/pinot/broker/broker/HelixBrokerStarterTest.java

```diff
@@ -145,10 +156,27 @@ public void testResourceAndTagAssignment() throws Exception {
     Assert.assertEquals(idealState.getInstanceSet(COFFEE_TABLE_NAME).size(), 6);
     Assert.assertEquals(idealState.getInstanceSet(DINING_TABLE_NAME).size(), 6);
 
-    Thread.sleep(3000);
+    // Wait up to 30s for broker external view to reach the expected size
+    waitForPredicate(new Callable<Boolean>() {
+      @Override
+      public Boolean call() throws Exception {
+        return _helixAdmin.getResourceExternalView(HELIX_CLUSTER_NAME, CommonConstants.Helix.BROKER_RESOURCE_INSTANCE)
+            .getStateMap(COFFEE_TABLE_NAME).size() == 6;
```

Comment:
Would be nice to have all the magic numbers replaced with constants, for readability.